unix_env/bb.org-overlays/tools/pinmux-generator/PocketBeagle_config-pin.txt

P1_01_PIN="power"
P1_01_INFO="VIN-AC"
P1_01_CAPE=""

P1_02_PRU="119"
P1_02_GPIO="87"
P1_02_PIN="gpio_input"
P1_02_PINMUX="default gpio gpio_pu gpio_pd gpio_input pruout pruin"
P1_02_INFO="gpio2_23 default gpio2_23 gpio2_23 gpio2_23 gpio2_23 pru1_out9 pru1_in9"
P1_02_CAPE=""

P1_03_PIN="system"
P1_03_INFO="usb1_vbus_out"
P1_03_CAPE=""

P1_04_PRU="121"
P1_04_GPIO="89"
P1_04_PIN="gpio"
P1_04_PINMUX="default gpio gpio_pu gpio_pd gpio_input pruout pruin"
P1_04_INFO="gpio2_25 default gpio2_25 gpio2_25 gpio2_25 gpio2_25 pru1_out11 pru1_in11"
P1_04_CAPE=""

P1_05_PIN="system"
P1_05_INFO="usb1_vbus_in"
P1_05_CAPE=""

P1_06_PRU="37"
P1_06_GPIO="5"
P1_06_PIN="spi_cs"
P1_06_PINMUX="default gpio gpio_pu gpio_pd gpio_input spi_cs i2c pwm pru_uart"
P1_06_INFO="spi0_cs0 default gpio0_5 gpio0_5 gpio0_5 gpio0_5 spi0_cs0 i2c1_scl ehrpwm0_synci pru_uart"
P1_06_CAPE=""

P1_07_PIN="system"
P1_07_INFO="VIN-USB"
P1_07_CAPE=""

P1_08_PRU="34"
P1_08_GPIO="2"
P1_08_PIN="spi_sclk"
P1_08_PINMUX="default gpio gpio_pu gpio_pd gpio_input spi_sclk uart i2c pwm pru_uart"
P1_08_INFO="spi0_sclk default gpio0_2 gpio0_2 gpio0_2 gpio0_2 spi0_sclk uart2_rxd i2c2_sda ehrpwm0a pru_uart"
P1_08_CAPE=""

P1_09_PIN="system"
P1_09_INFO="USB1-DN"
P1_09_CAPE=""

P1_10_PRU="35"
P1_10_GPIO="3"
P1_10_PIN="spi"
P1_10_PINMUX="default gpio gpio_pu gpio_pd gpio_input spi uart i2c pwm pru_uart"
P1_10_INFO="spi0_d0 default gpio0_3 gpio0_3 gpio0_3 gpio0_3 spi0_d0 uart2_txd i2c2_scl ehrpwm0b pru_uart"
P1_10_CAPE=""

P1_11_PIN="system"
P1_11_INFO="USB1-DP"
P1_11_CAPE=""

P1_12_PRU="36"
P1_12_GPIO="4"
P1_12_PIN="spi"
P1_12_PINMUX="default gpio gpio_pu gpio_pd gpio_input spi i2c pwm pru_uart"
P1_12_INFO="spi0_d1 default gpio0_4 gpio0_4 gpio0_4 gpio0_4 spi0_d1 i2c1_sda ehrpwm0_tripzone_input pru_uart"
P1_12_CAPE=""

P1_13_PIN="system"
P1_13_INFO="USB1-ID"
P1_13_CAPE=""

P1_14_PIN="power"
P1_14_INFO="VOUT-3.3V"
P1_14_CAPE=""

P1_15_PIN="gnd"
P1_15_INFO="GND"
P1_15_CAPE=""

P1_16_PIN="gnd"
P1_16_INFO="GND"
P1_16_CAPE=""

P1_17_PIN="power"
P1_17_INFO="VREFN"
P1_17_CAPE=""

P1_18_PIN="power"
P1_18_INFO="VREFP"
P1_18_CAPE=""

P1_19_PIN="adc"
P1_19_INFO="AIN0"
P1_19_CAPE=""

P1_20_PRU="52"
P1_20_GPIO="20"
P1_20_PIN="gpio"
P1_20_PINMUX="default gpio gpio_pu gpio_pd gpio_input pruin"
P1_20_INFO="gpio0_20 default gpio0_20 gpio0_20 gpio0_20 gpio0_20 pru0_in16"
P1_20_CAPE=""

P1_21_PIN="adc"
P1_21_INFO="AIN1"
P1_21_CAPE=""

P1_22_PIN="gnd"
P1_22_INFO="GND"
P1_22_CAPE=""

P1_23_PIN="adc"
P1_23_INFO="AIN2"
P1_23_CAPE=""

P1_24_PIN="power"
P1_24_INFO="VOUT-5V"
P1_24_CAPE=""

P1_25_PIN="adc"
P1_25_INFO="AIN3"
P1_25_CAPE=""

P1_26_PRU="44"
P1_26_GPIO="12"
P1_26_PIN="i2c"
P1_26_PINMUX="default gpio gpio_pu gpio_pd gpio_input spi_cs can i2c pru_uart"
P1_26_INFO="i2c2_sda default gpio0_12 gpio0_12 gpio0_12 gpio0_12 spi1_cs0 dcan0_tx i2c2_sda pru_uart"
P1_26_CAPE=""

P1_27_PIN="adc"
P1_27_INFO="AIN4"
P1_27_CAPE=""

P1_28_PRU="45"
P1_28_GPIO="13"
P1_28_PIN="i2c"
P1_28_PINMUX="default gpio gpio_pu gpio_pd gpio_input spi_cs can i2c pru_uart"
P1_28_INFO="i2c2_scl default gpio0_13 gpio0_13 gpio0_13 gpio0_13 spi1_cs1 dcan0_rx i2c2_scl pru_uart"
P1_28_CAPE=""

P1_29_PRU="149"
P1_29_GPIO="117"
P1_29_PIN="pruin"
P1_29_PINMUX="default gpio gpio_pu gpio_pd gpio_input qep pruout pruin"
P1_29_INFO="pru0_in7 default gpio3_21 gpio3_21 gpio3_21 gpio3_21 eqep0_strobe pru0_out7 pru0_in7"
P1_29_CAPE=""

P1_30_PRU="75"
P1_30_GPIO="43"
P1_30_PIN="uart"
P1_30_PINMUX="default gpio gpio_pu gpio_pd gpio_input spi_cs uart can i2c pruout pruin"
P1_30_INFO="uart0_txd default gpio1_11 gpio1_11 gpio1_11 gpio1_11 spi1_cs1 uart0_txd dcan0_rx i2c2_scl pru1_out15 pru1_in15"
P1_30_CAPE=""

P1_31_PRU="146"
P1_31_GPIO="114"
P1_31_PIN="pruin"
P1_31_PINMUX="default gpio gpio_pu gpio_pd gpio_input qep pruout pruin"
P1_31_INFO="pru0_in4 default gpio3_18 gpio3_18 gpio3_18 gpio3_18 eqep0a_in pru0_out4 pru0_in4"
P1_31_CAPE=""

P1_32_PRU="74"
P1_32_GPIO="42"
P1_32_PIN="uart"
P1_32_PINMUX="default gpio gpio_pu gpio_pd gpio_input spi_cs uart can i2c pruout pruin"
P1_32_INFO="uart0_rxd default gpio1_10 gpio1_10 gpio1_10 gpio1_10 spi1_cs0 uart0_rxd dcan0_tx i2c2_sda pru1_out14 pru1_in14"
P1_32_CAPE=""

P1_33_PRU="143"
P1_33_GPIO="111"
P1_33_PIN="pruin"
P1_33_PINMUX="default gpio gpio_pu gpio_pd gpio_input spi pwm pruout pruin"
P1_33_INFO="pru0_in1 default gpio3_15 gpio3_15 gpio3_15 gpio3_15 spi1_d0 ehrpwm0b pru0_out1 pru0_in1"
P1_33_CAPE=""

P1_34_PRU="58"
P1_34_GPIO="26"
P1_34_PIN="gpio"
P1_34_PINMUX="default gpio gpio_pu gpio_pd gpio_input pwm"
P1_34_INFO="gpio0_26 default gpio0_26 gpio0_26 gpio0_26 gpio0_26 ehrpwm2_tripzone_input"
P1_34_CAPE=""

P1_35_PRU="120"
P1_35_GPIO="88"
P1_35_PIN="pruin"
P1_35_PINMUX="default gpio gpio_pu gpio_pd gpio_input pruout pruin"
P1_35_INFO="pru1_in10 default gpio2_24 gpio2_24 gpio2_24 gpio2_24 pru1_out10 pru1_in10"
P1_35_CAPE=""

P1_36_PRU="142"
P1_36_GPIO="110"
P1_36_PIN="pwm"
P1_36_PINMUX="default gpio gpio_pu gpio_pd gpio_input spi_sclk pwm pruout pruin"
P1_36_INFO="ehrpwm0a default gpio3_14 gpio3_14 gpio3_14 gpio3_14 spi1_sclk ehrpwm0a pru0_out0 pru0_in0"
P1_36_CAPE=""

P2_01_PRU="82"
P2_01_GPIO="50"
P2_01_PIN="pwm"
P2_01_PINMUX="default gpio gpio_pu gpio_pd gpio_input pwm"
P2_01_INFO="ehrpwm1a default gpio1_18 gpio1_18 gpio1_18 gpio1_18 ehrpwm1a"
P2_01_CAPE=""

P2_02_PRU="91"
P2_02_GPIO="59"
P2_02_PIN="gpio"
P2_02_PINMUX="default gpio gpio_pu gpio_pd gpio_input"
P2_02_INFO="gpio1_27 default gpio1_27 gpio1_27 gpio1_27 gpio1_27"
P2_02_CAPE=""

P2_03_PRU="55"
P2_03_GPIO="23"
P2_03_PIN="gpio"
P2_03_PINMUX="default gpio gpio_pu gpio_pd gpio_input pwm"
P2_03_INFO="gpio0_23 default gpio0_23 gpio0_23 gpio0_23 gpio0_23 ehrpwm2b"
P2_03_CAPE=""

P2_04_PRU="90"
P2_04_GPIO="58"
P2_04_PIN="gpio"
P2_04_PINMUX="default gpio gpio_pu gpio_pd gpio_input"
P2_04_INFO="gpio1_26 default gpio1_26 gpio1_26 gpio1_26 gpio1_26"
P2_04_CAPE=""

P2_05_PRU="62"
P2_05_GPIO="30"
P2_05_PIN="uart"
P2_05_PINMUX="default gpio gpio_pu gpio_pd gpio_input uart"
P2_05_INFO="uart4_rxd default gpio0_30 gpio0_30 gpio0_30 gpio0_30 uart4_rxd"
P2_05_CAPE=""

P2_06_PRU="89"
P2_06_GPIO="57"
P2_06_PIN="gpio"
P2_06_PINMUX="default gpio gpio_pu gpio_pd gpio_input"
P2_06_INFO="gpio1_25 default gpio1_25 gpio1_25 gpio1_25 gpio1_25"
P2_06_CAPE=""

P2_07_PRU="63"
P2_07_GPIO="31"
P2_07_PIN="uart"
P2_07_PINMUX="default gpio gpio_pu gpio_pd gpio_input uart"
P2_07_INFO="uart4_txd default gpio0_31 gpio0_31 gpio0_31 gpio0_31 uart4_txd"
P2_07_CAPE=""

P2_08_PRU="92"
P2_08_GPIO="60"
P2_08_PIN="gpio"
P2_08_PINMUX="default gpio gpio_pu gpio_pd gpio_input"
P2_08_INFO="gpio1_28 default gpio1_28 gpio1_28 gpio1_28 gpio1_28"
P2_08_CAPE=""

P2_09_PRU="47"
P2_09_GPIO="15"
P2_09_PIN="i2c"
P2_09_PINMUX="default gpio gpio_pu gpio_pd gpio_input uart can i2c pru_uart pruin"
P2_09_INFO="i2c1_scl default gpio0_15 gpio0_15 gpio0_15 gpio0_15 uart1_txd dcan1_rx i2c1_scl pru_uart pru0_in16"
P2_09_CAPE=""

P2_10_PRU="84"
P2_10_GPIO="52"
P2_10_PIN="gpio"
P2_10_PINMUX="default gpio gpio_pu gpio_pd gpio_input qep"
P2_10_INFO="gpio1_20 default gpio1_20 gpio1_20 gpio1_20 gpio1_20 eqep1a_in"
P2_10_CAPE=""

P2_11_PRU="46"
P2_11_GPIO="14"
P2_11_PIN="i2c"
P2_11_PINMUX="default gpio gpio_pu gpio_pd gpio_input uart can i2c pru_uart pruin"
P2_11_INFO="i2c1_sda default gpio0_14 gpio0_14 gpio0_14 gpio0_14 uart1_rxd dcan1_tx i2c1_sda pru_uart pru1_in16"
P2_11_CAPE=""

P2_12_PIN="system"
P2_12_INFO="POWER_BUTTON"
P2_12_CAPE=""

P2_13_PIN="power"
P2_13_INFO="VOUT-5V"
P2_13_CAPE=""

P2_14_PIN="power"
P2_14_INFO="BAT-VIN"
P2_14_CAPE=""

P2_15_PIN="gnd"
P2_15_INFO="GND"
P2_15_CAPE=""

P2_16_PIN="power"
P2_16_INFO="BAT-TEMP"
P2_16_CAPE=""

P2_17_PRU="97"
P2_17_GPIO="65"
P2_17_PIN="gpio"
P2_17_PINMUX="default gpio gpio_pu gpio_pd gpio_input"
P2_17_INFO="gpio2_1 default gpio2_1 gpio2_1 gpio2_1 gpio2_1"
P2_17_CAPE=""

P2_18_PRU="79"
P2_18_GPIO="47"
P2_18_PIN="gpio"
P2_18_PINMUX="default gpio gpio_pu gpio_pd gpio_input qep pru_ecap pruin"
P2_18_INFO="gpio1_15 default gpio1_15 gpio1_15 gpio1_15 gpio1_15 eqep2_strobe pru_ecap pru0_in15"
P2_18_CAPE=""

P2_19_PRU="59"
P2_19_GPIO="27"
P2_19_PIN="gpio"
P2_19_PINMUX="default gpio gpio_pu gpio_pd gpio_input pwm"
P2_19_INFO="gpio0_27 default gpio0_27 gpio0_27 gpio0_27 gpio0_27 ehrpwm0_synco"
P2_19_CAPE=""

P2_20_PRU="96"
P2_20_GPIO="64"
P2_20_PIN="gpio"
P2_20_PINMUX="default gpio gpio_pu gpio_pd gpio_input"
P2_20_INFO="gpio2_0 default gpio2_0 gpio2_0 gpio2_0 gpio2_0"
P2_20_CAPE=""

P2_21_PIN="gnd"
P2_21_INFO="GND"
P2_21_CAPE=""

P2_22_PRU="78"
P2_22_GPIO="46"
P2_22_PIN="gpio"
P2_22_PINMUX="default gpio gpio_pu gpio_pd gpio_input qep pruin"
P2_22_INFO="gpio1_14 default gpio1_14 gpio1_14 gpio1_14 gpio1_14 eqep2_index pru0_in14"
P2_22_CAPE=""

P2_23_PIN="power"
P2_23_INFO="VOUT-3.3V"
P2_23_CAPE=""

P2_24_PRU="76"
P2_24_GPIO="44"
P2_24_PIN="gpio"
P2_24_PINMUX="default gpio gpio_pu gpio_pd gpio_input qep pruout"
P2_24_INFO="gpio1_12 default gpio1_12 gpio1_12 gpio1_12 gpio1_12 eqep2a_in pru0_out14"
P2_24_CAPE=""

P2_25_PRU="73"
P2_25_GPIO="41"
P2_25_PIN="spi"
P2_25_PINMUX="default gpio gpio_pu gpio_pd gpio_input spi spi_cs uart can i2c"
P2_25_INFO="spi1_d1 default gpio1_9 gpio1_9 gpio1_9 gpio1_9 spi1_d1 spi1_cs0 uart4_txd dcan1_rx i2c1_scl"
P2_25_CAPE=""

P2_26_PIN="system"
P2_26_INFO="RESET#"
P2_26_CAPE=""

P2_27_PRU="72"
P2_27_GPIO="40"
P2_27_PIN="spi"
P2_27_PINMUX="default gpio gpio_pu gpio_pd gpio_input spi uart can i2c"
P2_27_INFO="spi1_d0 default gpio1_8 gpio1_8 gpio1_8 gpio1_8 spi1_d0 uart4_rxd dcan1_tx i2c1_sda"
P2_27_CAPE=""

P2_28_PRU="148"
P2_28_GPIO="116"
P2_28_PIN="pruin"
P2_28_PINMUX="default gpio gpio_pu gpio_pd gpio_input qep pruout pruin"
P2_28_INFO="pru0_in6 default gpio3_20 gpio3_20 gpio3_20 gpio3_20 eqep0_index pru0_out6 pru0_in6"
P2_28_CAPE=""

P2_29_PRU="39"
P2_29_GPIO="7"
P2_29_PIN="spi_sclk"
P2_29_PINMUX="default gpio gpio_pu gpio_pd gpio_input spi_cs spi_sclk uart pwm pru_ecap"
P2_29_INFO="spi1_sclk default gpio0_7 gpio0_7 gpio0_7 gpio0_7 spi1_cs1 spi1_sclk uart3_txd ecap0_in_pwm0_out pru_ecap"
P2_29_CAPE=""

P2_30_PRU="145"
P2_30_GPIO="113"
P2_30_PIN="pruin"
P2_30_PINMUX="default gpio gpio_pu gpio_pd gpio_input spi_cs pwm pruout pruin"
P2_30_INFO="pru0_in3 default gpio3_17 gpio3_17 gpio3_17 gpio3_17 spi1_cs0 ehrpwm0_synci pru0_out3 pru0_in3"
P2_30_CAPE=""

P2_31_PRU="51"
P2_31_GPIO="19"
P2_31_PIN="spi_cs"
P2_31_PINMUX="default gpio gpio_pu gpio_pd gpio_input spi_cs pruin"
P2_31_INFO="spi1_cs1 default gpio0_19 gpio0_19 gpio0_19 gpio0_19 spi1_cs1 pru1_in16"
P2_31_CAPE=""

P2_32_PRU="144"
P2_32_GPIO="112"
P2_32_PIN="pruin"
P2_32_PINMUX="default gpio gpio_pu gpio_pd gpio_input spi pwm pruout pruin"
P2_32_INFO="pru0_in2 default gpio3_16 gpio3_16 gpio3_16 gpio3_16 spi1_d1 ehrpwm0_tripzone_input pru0_out2 pru0_in2"
P2_32_CAPE=""

P2_33_PRU="77"
P2_33_GPIO="45"
P2_33_PIN="gpio"
P2_33_PINMUX="default gpio gpio_pu gpio_pd gpio_input qep pruout"
P2_33_INFO="gpio1_13 default gpio1_13 gpio1_13 gpio1_13 gpio1_13 eqep2b_in pru0_out15"
P2_33_CAPE=""

P2_34_PRU="147"
P2_34_GPIO="115"
P2_34_PIN="pruin"
P2_34_PINMUX="default gpio gpio_pu gpio_pd gpio_input qep pruout pruin"
P2_34_INFO="pru0_in5 default gpio3_19 gpio3_19 gpio3_19 gpio3_19 eqep0b_in pru0_out5 pru0_in5"
P2_34_CAPE=""

P2_35_PRU="118"
P2_35_GPIO="86"
P2_35_PIN="gpio_input"
P2_35_PINMUX="default gpio gpio_pu gpio_pd gpio_input pruout pruin"
P2_35_INFO="gpio2_22 default gpio2_22 gpio2_22 gpio2_22 gpio2_22 pru1_out8 pru1_in8"
P2_35_CAPE=""


CHANGE_LOG
CHANGE_LOG

For the Github commit log see here: https://github.com/jks-prv/Beagle_SDR_GPS/commits/master

v1.175  Mar 5, 2018
    GPS fixes and improvements.
    Include copy of bb-cape-overlay package for Debian 8.10 /dev/spidev* workaround

v1.174  Feb 23, 2018
    Display DHCP/static IP address on Beagle LEDs.
        See kiwisdr.com/quickstart/index.html#id-leds

v1.173  Feb 14, 2018
    Check against bad GPS az/el solutions.

v1.172  Feb 13, 2018
    Respect GPS subframe alert flag. Don't include sat in position fix if alert flag is set.
    Add "include alerted?" switch to GPS tab on admin page to override alert exclusion.

v1.171  Feb 11, 2018
    Admin GPS tab additions.
        Shadow map. Shows long term areas of non-reception caused by "shadowing" of buildings and terrain.
        For Asia/Oceania, a locator for the new geosynchronous QZS-3 sat will appear as a circle.
            Combined with the shadow map this will indicate if QZS-3 is likely to be received.

v1.170  Feb 8, 2018
    Added GPS support for Japan's Quasi-Zenith Satellite System (QZSS).
        Should have reception over entire Asia-Oceania region (heard fine in New Zealand).

v1.169  Feb 6, 2018
    Faster GPS sat acquisition (260 -> 60 msec) from Christoph Mayer.
        Adjust bin size so power-of-two FFTs can be used.
        Use ARM NEON SIMD instructions.

v1.168  Feb 1, 2018
    Improve speed of DX label handling.

v1.167  Jan 31, 2018
    FSK:
        Leave zoom alone when changing shift etc.
        Fix bug in EFR mode that makes it print “undefined”.

v1.166  Jan 30, 2018
    FSK: Add shift position markers on top of waterfall below passband.
    WSPR: darken call/grid link colors.

v1.165  Jan 28, 2018
    Add alternate 80m and 60m WSPR frequencies.
    Fix bug preventing inactive connection timeout from working.
    Add custom shift and baud rate fields to FSK extension.

v1.164  Jan 27, 2018
    Catch null pointer when kiwisdr.com is down.
    On admin sdr.hu tab, limit resolution of GPS assist lat/lon field-fill for privacy reasons. 

v1.163  Jan 23, 2018
    FSK extension:
        EFR Teleswitch decoding improvements.
        CHU decoding working.

v1.162  Jan 22, 2018
    FSK extension: support for EFR Teleswitch stations (130-140 kHz in Europe).

v1.161  Jan 21, 2018
    Actually enable the FSK extension.
        Add more baud and shift values to menus.
        Improvements to the framing tool.

v1.160  Jan 20, 2018
    Fixed FSK menu bug.

v1.159  Jan 20, 2018
    Navtex: add frequency to DX mode, next/prev menu entry buttons.
    Added FSK extension. A work in progress. Not all framing modes seem to work.

v1.158  Jan 17, 2018
    Add Navtex DX mode. Each new message prints a single line containing a timestamp,
        header information and the first 80 characters of the message.

v1.157  Jan 16, 2018
    Fix user display on admin status tab
    Add volume URL parameter, e.g. kiwisdr:8073/?vol=100
        vol = 0 to 200(%), default is 50%

v1.156  Jan 12, 2018
    Fixed FAX frequencies: Athens (way off) and Northwood UK 4 MHz (100 Hz error).
    Added GPS tracking PLL IQ display to admin interface.
    Added scope display to Navtex extension.

v1.155  Jan 3, 2018
    Navtex extension:
        Add baud error indicator.
        Add HF frequencies menu.
        Text window clear button.
        Mark/space invert checkbox.
    Another fix to NDB database lookup function.
    API fix for antenna switch extension.

v1.154  Jan 2, 2018
    New Navtex extension.
    Major user interface changes:
        New “option tab” user interface for lower right control panel.
        Removed lower left status panel. Replaced with the new "Users" and "Stats" option tabs.
        Next/previous history buttons next to frequency entry box.
            If using a keyboard press return key to select displayed frequency.
            Otherwise prior frequency will return after three seconds.
            Doesn't really work yet for mobile devices.
        Added numeric dBm value to S-meter.
        Added classic LMS denoiser and autonotch to audio tab. Filter controls on extension panel.
            The LMS filters are not a panacea for all filtering problems.
            They are tricky to adjust and don't work in all situations.
            The presets are a work-in-progress.
        Added audio mute icon/button to right of spectrum button (renamed "Spec" to fit).
        WF option tab
            Number after "WF" on option tab button is current zoom level, e.g. "WF10"
            "Auto Scale" button: automatically sets WF min/max.
                Best when there are a range of signal levels.
            "Slow Dev" button: Use when you have audio drops on slow mobile devices (e.g. old iPad).
                Uses single color in spectrum and slower waterfall speed.
            "colormap" & "contrast" menus: beginnings of waterfall colormap control. Work-in-progress.
            "spec filter" menu: variable spectrum time constant so decay time can be varied.
        Add URL parameter to set initial waterfall speed:
            wf={0, off, 1, 1hz, s, slow, m, med, f, fast} e.g. kiwi:8073/?wf=1hz
        Fixed problem with NDB database lookup not working.
        Removed “less buffering” and “compression” buttons from Audio tab until they are working.

v1.153  Dec 7, 2017
    Add threshold slider to noise blanker.

v1.152  Dec 6, 2017
    First attempt at adding a noise blanker.
    Added variable speed waterfall ("WF rate" slider)
    Now on extended control panel, via "more" button:
        The "WF max" slider.

code/web/config.js
var init_freq = 1107000;
var init_mode = 'am';
var init_zoom = 0;
var init_max_dB = 0;			// -25 -30 10 -35
var init_min_dB = -120;		// -105 -300 -110 -95

// ITU regions:
// 1 = Europe, Africa
// 2 = North & South America
// 3 = Asia / Pacific

var svc = {
//	N: { name:'MHRS NoN',					color:'yellow' },
	B: { name:'Broadcast',					color:'red' },
	U: { name:'Utility', 					color:'green' },
	A: { name:'Amateur', 					color:'blue' },
	I: { name:'Industrial/Scientific', 	color:'orange' },
	M: { name:'Markers', 					color:'purple' },
};

var bands=[];

// for band scale:
//		'*' means applies to all regions vs '1', '2' or '3'
//		first char '>' chooses which region to use when multiple regions present

// for "select band" menu:
//		only region '*', '>' or 'm' entries are used (e.g. not ISM that uses '-')
//		a region of 'm' appears in band menu only -- never in band scale
//		if selected freq/mode ('sel:') not specified goes to middle of band

// MHRS NoN
//bands.push({ s:svc.N, min:426, max:500, sel:"500cwn", region:"m", name:"500 kc" });
//bands.push({ s:svc.N, min:6383, max:6478, sel:"6477.5cwn", region:"m", name:"6 Mc" });
//bands.push({ s:svc.N, min:8438, max:8658, sel:"8642cwn", region:"m", name:"8 Mc" });
//bands.push({ s:svc.N, min:12695, max:12993, sel:"12808.5cwn", region:"m", name:"12 Mc" });
//bands.push({ s:svc.N, min:16914, max:17221, sel:"17016.8cwn", region:"m", name:"16 Mc" });
//bands.push({ s:svc.N, min:22477, max:22478, sel:"22477.5", region:"m", name:"22 Mc" });

bands.push({ s:svc.B, min:153, max:279, chan:9, region:'E', name:"LW" });
bands.push({ s:svc.B, min:153, max:198, chan:9, region:'>', name:"LW" });	// stopped at 198 for local NDB band
bands.push({ s:svc.B, min:531, max:1602, sel:"1107am", chan:9, region:">E", name:"MW" });
bands.push({ s:svc.B, min:540, max:1700, sel:"1107am", chan:10, region:"U", name:"MW" });
//foo
//bands.push({ s:svc.B, min:1107, max:1107, sel:"1107am", chan:10, region:">", name:"test" });

bands.push({ s:svc.B, min:2300, max:2495, region:'*', name:"120m" });
bands.push({ s:svc.B, min:3200, max:3400, region:'*', name:"90m" });
bands.push({ s:svc.B, min:3900, max:4000, region:'*', name:"75m" });
bands.push({ s:svc.B, min:4750, max:5060, region:'*', name:"60m" });
bands.push({ s:svc.B, min:5900, max:6200, region:'*', name:"49m" });
bands.push({ s:svc.B, min:7200, max:7450, region:"13", name:"41m" });
bands.push({ s:svc.B, min:7300, max:7450, region:">2", name:"41m" });
bands.push({ s:svc.B, min:9400, max:9900, region:'*', name:"31m" });
bands.push({ s:svc.B, min:11600, max:12100, region:'*', name:"25m" });
bands.push({ s:svc.B, min:13570, max:13870, region:'*', name:"22m" });
bands.push({ s:svc.B, min:15100, max:15800, region:'*', name:"19m" });
bands.push({ s:svc.B, min:17480, max:17900, region:'*', name:"16m" });
bands.push({ s:svc.B, min:18900, max:19020, region:'*', name:"15m" });
bands.push({ s:svc.B, min:21450, max:21850, region:'*', name:"13m" });
bands.push({ s:svc.B, min:25600, max:26100, region:'*', name:"11m" });

bands.push({ s:svc.U, min:10, max:30, sel:"15.25lsb", region:'*', name:"VLF" });
bands.push({ s:svc.U, min:31, max:100, sel:"60cwn", region:'*', name:"LF" });
bands.push({ s:svc.U, min:200, max:472, sel:"346amn", region:">", name:"NDB" });	// local definition of NDB band
bands.push({ s:svc.U, min:283.5, max:472, region:"E", name:"NDB" });	// don't collide w/ LW below
bands.push({ s:svc.U, min:190, max:535, region:"U", name:"NDB" });
bands.push({ s:svc.U, min:294, max:320, sel:"308cw", region:"m", name:"DGPS" });
bands.push({ s:svc.U, min:2500, max:2500, sel:"2500amn", region:'*', name:"Time 2.5" });
bands.push({ s:svc.U, min:5000, max:5000, sel:"5000amn", region:'*', name:"Time 5" });
bands.push({ s:svc.U, min:10000, max:10000, sel:"10000amn", region:'*', name:"Time 10" });
bands.push({ s:svc.U, min:15000, max:15000, sel:"15000amn", region:'*', name:"Time 15" });
bands.push({ s:svc.U, min:20000, max:20000, sel:"20000amn", region:'*', name:"Time 20" });

bands.push({ s:svc.A, min:135.7, max:137.8, region:'*', name:"LF" });	// 2200m
bands.push({ s:svc.A, min:472, max:479, region:">23", name:"MF" });		// 630m
bands.push({ s:svc.A, min:1810, max:2000, region:"1", name:"160m" });
bands.push({ s:svc.A, min:1800, max:2000, region:">23", name:"160m" });
bands.push({ s:svc.A, min:3500, max:3800, region:"1", name:"80m" });
bands.push({ s:svc.A, min:3500, max:4000, region:">2", name:"80m" });
bands.push({ s:svc.A, min:3500, max:3900, region:"3", name:"80m" });
bands.push({ s:svc.A, min:5250, max:5450, region:'*', name:"60m" });	// many variations
bands.push({ s:svc.A, min:7000, max:7200, region:"1", name:"40m" });
bands.push({ s:svc.A, min:7000, max:7300, region:">23", name:"40m" });
bands.push({ s:svc.A, min:10100, max:10150, region:">12", name:"30m" });
bands.push({ s:svc.A, min:10100, max:10157.3, region:"3", name:"30m" });
bands.push({ s:svc.A, min:14000, max:14350, region:'*', name:"20m" });
bands.push({ s:svc.A, min:18068, max:18168, region:'*', name:"17m" });
bands.push({ s:svc.A, min:21000, max:21450, region:'*', name:"15m" });
bands.push({ s:svc.A, min:24890, max:24990, region:'*', name:"12m" });
bands.push({ s:svc.A, min:28000, max:29700, region:'*', name:"10m" });

bands.push({ s:svc.I, min:6765, max:6795, region:"-", name:"ISM" });
bands.push({ s:svc.I, min:13553, max:13567, region:"-", name:"ISM" });
bands.push({ s:svc.I, min:26957, max:27283, region:"-", name:"ISM" });

bands.push({ s:svc.M, min:3594.4, max:3594.4, sel:"3594.4cwn", region:'m', name:"3594" });
bands.push({ s:svc.M, min:4558.4, max:4558.4, sel:"4558.4cwn", region:'m', name:"4558" });
bands.push({ s:svc.M, min:5154.4, max:5154.4, sel:"5154.4cwn", region:'m', name:"5154" });
bands.push({ s:svc.M, min:5156.8, max:5156.8, sel:"5156.8cwn", region:'m', name:"5156 L" });
bands.push({ s:svc.M, min:5292.0, max:5292.0, sel:"5292.0cwn", region:'m', name:"5292 B" });
bands.push({ s:svc.M, min:6928.0, max:6928.0, sel:"6928.0cwn", region:'m', name:"6928 V" });
bands.push({ s:svc.M, min:7039.4, max:7039.4, sel:"7039.4cwn", region:'m', name:"7039" });
bands.push({ s:svc.M, min:7509.2, max:7509.2, sel:"7509.2cwn", region:'m', name:"7509" });
bands.push({ s:svc.M, min:8000.0, max:8000.0, sel:"8000.0cwn", region:'m', name:"8000 C" });
bands.push({ s:svc.M, min:8495.3, max:8495.3, sel:"8495.3cwn", region:'m', name:"8495" });
bands.push({ s:svc.M, min:10872.2, max:10872.2, sel:"10872.2cwn", region:'m', name:"10872" });
bands.push({ s:svc.M, min:13528.4, max:13528.4, sel:"13528.4cwn", region:'m', name:"13528" });
bands.push({ s:svc.M, min:16332.0, max:16332.0, sel:"16332.0cwn", region:'m', name:"16332" });
bands.push({ s:svc.M, min:20048.4, max:20048.4, sel:"20048.4cwn", region:'m', name:"20048" });
